tutorials/docs/101_fusion.md

```diff
@@ -0,0 +1,106 @@
+finished before returning control. However, when a resource is used to generate
+another one, the next create call will wait for the origin resource to be
+finished before starting the creation process (e.g, `(create-dataset source-id)`
+will wait for the source to be finished before creating the dataset from it).
+
+If you want to create predictions for many new inputs, you can do so by
+creating
+a `batchprediction` resource. First, you will need to upload to the platform
+all the input data that you want to predict for and create the corresponding
+`source` and `dataset` resources. In the example, we'll be assuming you already
+created a `fusion` following the steps 0 to 2 in the previous snippet.
+
+```
+    ;; step 3: creating a source from the data in your remote "https://static.bigml.com/csv/test_iris.csv" file
+    (define test-source-id (create-source {"remote" "https://static.bigml.com/csv/test_iris.csv"))
+    ;; step 4: creating a dataset from the previously created `source`
+    (define test-dataset-id (create-dataset test-source-id))
+    ;; step 5: creating a batch prediction
+    (define batch-prediction-id (create-batchprediction fusion-id test-dataset-id))
+```
+
+The batch prediction output (as well as any of the resources created)
+can be configured using additional arguments in the corresponding create calls.
+For instance, to include all the information in the original dataset in the
+output you would change `step 5` to:
+
+```
+    (define batch-prediction-id (create-batchprediction
+                                  {"model" fusion-id
+                                   "dataset" test-dataset-id
+                                   "all_fields" true}))
+```
+
+and to generate a new dataset that adds the prediction as a new column appended
+at the end of the original ones and retrieve the information therein, the step
+would be:
+
+```
+    ;; step 5: creating a batch prediction. Note that we now use
+    ;; `create-and-wait-batchprediction` to ensure that we wait for the
+    ;; resource to be finished to read its properties in the next step.
+    ;; We can also use `create-batchprediction` and wait later for the
+    ;; resource to finish using  `wait`. The `create-[resource-type]` and
+    ;; `create-and-wait-[resource-type]` procedures are available for
+    ;; all resources
+    (define batch-prediction-id (create-and-wait-batchprediction
+                                  {"model" fusion-id
+                                   "dataset" test-dataset-id
+                                   "all_fields" true
+                                   "output_dataset" true}))
+    ;; step 6: retrieving the batch prediction information
+    (define batch-prediction-resource (fetch batch-prediction-id))
+    ;; step 7: extracting the output dataset ID and waiting for it to be
+    ;; finished too in order to use it.
+    (define batch-prediction-ds
+      (wait (batch-prediction-resource "output_dataset_resource")))
```

Comment:
I've used the log-info function in the first example of each 101. Mainly because I suggest that they try it on the REPL and thought it might be useful to leave a trace of what was happening. The rest of code are extensions and I thought too that was fine with no logs at all.